### PR TITLE
Tests for the new serialization functions

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/basic/fileio.c [hunk 1]
```diff
@@ -1221,6 +1221,22 @@ int mkostemp_safe(char *pattern) {
...
+        int fd;
+        FILE *f;
+
+        fd = mkostemp_safe(pattern);
+        if (fd < 0)
+                return fd;
+
+        f = fdopen(fd, mode);
+        if (!f)
+                return -errno;
```
Comment (poettering) [inline src/basic/fileio.c:1235]: error path leaks `fd`

**Comment by poettering** [comment]
lgtm, but the error path thing should be fixed...

**Comment by poettering** [comment]
`cc1: out of memory allocating 107374688 bytes after a total of 138547200 bytes`

**Comment by poettering** [comment]
I figure the `million()` isn't really liked by the gcc version there on the ubuntu CI...

**Comment by keszybz** [comment]
Updated. Let's see if the CI passes.

**Comment by poettering** [comment]
s390 passed, let's see if i386 and arm64 (as the more resource constrained systems) will pass too

**Comment by poettering** [comment]
amd64 passed too. let's wait for the other two, and merge then

**Comment by poettering** [comment]
yippieh! all green! merging.

**Pull request merged**